user/src/com/google/web/bindery/autobean/vm/impl/ProxyAutoBean.java

```diff
@@ -112,34 +112,37 @@ private static Map<String, Data> calculateData(Class<?> beanType) {
       if (toReturn == null) {
         Map<String, Data> getters = new HashMap<String, Data>();
         List<Method> setters = new ArrayList<Method>();
-        for (Method method : beanType.getMethods()) {
-          if (BeanMethod.GET.matches(method)) {
-            // match methods on their name for now, to find the most specific
-            // override
-            String name = method.getName();
+        if (!(Collection.class.isAssignableFrom(beanType)
+            || Map.class.isAssignableFrom(beanType))) {
```

Comment:
I think this should be
```suggestion
        if (!(Collection.class.isAssignableFrom(beanType)
            && !Map.class.isAssignableFrom(beanType))) {
```
so that we don't run this code with either maps or collections (since this is explicitly for beans)?

Reply:
Ignore, my comment is wrong.